Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -1,34 +1,223 @@
+///
+/// We adopt the conventions in [1] and [2] for the tangent space of this manifold and for
+/// derivatives of functions to/from this manifold. In particular:
+///
+///   1. The tangent space at every point is the Lie algebra so(2).
+///   2. For Lie groups `G`, `H`, with algebras `g`, `h`, the differential of any differentiable
+///      function `f : G -> H` at `a \in G` is the linear map `df_a : g -> h` such that
+///        f(a * exp(eps)) ~= f(a) * exp(df_a(eps))
+///      where `eps` is a "small" element of `g`. (And the pullback is the dual map of `df_a`).
+///
+/// (1) is accomplished simply by defining the `TangentVector` type of `Pose2` to be so(2).
+///
+/// (2) is accomplished for `Pose2` by defining differentials[3] satisfying (2) for three
+//  functions:
+///   - .t: SE(2) -> R^2
+///   - .rot: SE(2) -> SO(2)
+///   - init: R^2 x SO(2) -> SE(2)
+/// Since these are the only functions exposed by the original `Pose2` declaration, all other
+/// functions involving `Pose2` are forced to be compositions of these. Therefore, the derivatives
+/// synthesized by Swift's automatic differentiation transform satisfy (2).
+///
+/// [1]: https://github.com/borglab/gtsam/blob/develop/doc/LieGroups.pdf
+/// [2]: https://github.com/borglab/gtsam/blob/develop/doc/math.pdf
+/// [3]: Actually, we define the pullbacks because Swift doesn't support differentials very well
+///      yet.
+public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
+  /// The pose's translation.
+  ///
+  /// This is the projection SE(2) -> R^2. (Note: not a group homomorphism!)
+  @differentiable public var t: Point2 { tStorage }
+
+  /// The pose's rotation.
+  ///
+  /// This is the projection SE(2) -> SO(2).
+  @differentiable public var rot: Rot2 { rotStorage }
+
+  /// Creates a `Pose2` with rotation `r` and translation `t`.
+  ///
+  /// This is the bijection SO(2) x R^2 -> SE(2), where "x" means direct product of groups. (Note:
+  /// not a group homomorphism!)
   @differentiable
-  public init(_ x: Double, _ y: Double, _ theta: Double) {
-    t_ = Point2(x, y)
-    rot_ = Rot2(theta)
+  public init(_ t: Point2, _ r: Rot2) {
+    tStorage = t
+    rotStorage = r
+  }
+
+  /// The derivative of `init` satisfiying convention (2) described above.
+  @derivative(of: init(_:_:))
+  @usableFromInline
+  static func vjpInit(_ t: Point2, _ r: Rot2)
+    -> (value: Pose2, pullback: (TangentVector) -> (Point2.TangentVector, Rot2.TangentVector))
+  {
+    // Explanation of this calculation:
+    //
+    // We would like the differential of `init` at `(t, r)`, `dinit_(t, r)`, to satisfy:
+    //   init(t * exp(eps_t), r * exp(eps_r)) ~= init(t, r) * exp(dinit_(t, r)(eps_t, eps_r))
+    // where `eps_t`, `eps_r` are "small" elements of R^2 and so(2) respectively.
+    //
+    // Multiplying on the left by `init(t, r)^-1` gives
+    //   exp(dinit_(t, r)(eps_t, eps_r))
+    //     = init(t, r)^-1 * init(t * exp(eps_t), r * exp(eps_r))
```

Comment:
I'd state `t * exp(eps_t) == t + eps_t` and `inverse(r) = r^t` and use that
Also, maybe use 
eps_t -> v
eps_r -> w
to make this easier to parse.